lib/spl-build.js

```diff
@@ -423,33 +441,50 @@ export default {
   },
 
   shouldShowMenuSpl(event) {
+    if (atom.workspace.getActivePane().activeItem.constructor.name === 'TextEditor') {
+      return !!atom.workspace.getActiveTextEditor().getTitle().toLowerCase().endsWith('spl');
+    }
     return !!event.target.innerText.toLowerCase().endsWith('.spl');
   },
 
   shouldShowMenuMake(event) {
+    if (atom.workspace.getActivePane().activeItem.constructor.name === 'TextEditor') {
+      return atom.workspace.getActiveTextEditor().getTitle().toLowerCase() === 'makefile';
+    }
     return event.target.innerText.toLowerCase() === 'makefile';
   },
-
   shouldShowMenuSubmit(event) {
+    if (atom.workspace.getActivePane().activeItem.constructor.name === 'TextEditor') {
+      return !!atom.workspace.getActiveTextEditor().getTitle().toLowerCase().endsWith('.sab');
+    }
     return !!event.target.innerText.toLowerCase().endsWith('.sab');
   },
-
   shouldShowMenu(event) {
+    if (atom.workspace.getActivePane().activeItem.constructor.name === 'TextEditor') {
+      return !!(atom.workspace.getActiveTextEditor().getTitle().toLowerCase() === 'makefile'
+        || atom.workspace.getActiveTextEditor().getTitle().toLowerCase().endsWith('.spl')
+        || atom.workspace.getActiveTextEditor().getTitle().toLowerCase().endsWith('.sab'));
+    }
     return !!(event.target.innerText.toLowerCase() === 'makefile'
       || event.target.innerText.toLowerCase().endsWith('.spl')
       || event.target.innerText.toLowerCase().endsWith('.sab'));
   },
-
   shouldShowMenuV4(event) {
+    if (atom.workspace.getActivePane().activeItem.constructor.name === 'TextEditor') {
+      this.shouldShowMenu = this.shouldShowMenu.bind(this);
```

Comment:
the bind() calls in shouldShowMenuV* functions are not needed, the bind() where the menu is defined sets 'this', so calls down that chain have the correct context for 'this'. Can remove the added bind() and existing bind() calls in shouldShowMenuV4() and shouldShowMenuV5(). Because of that, and the other recommended change to check using getActiveTextEditor(), you can refactor those two functions to simply do
`return this.shouldShowMenu(event) && this.apiVersion === CONF_API_VERSION_V4;`
and
`return this.shouldShowMenu(event) && this.apiVersion === CONF_API_VERSION_V5;`
respectively.